Add MockTcpServer, DaytimeTCPClient integration tests

jkbkupczyk opened the pull request:
- Adds MockTcpServer - base class for all other TCP Server impls, e.g. we can have _MockDiscardTCPServer_ to test _DiscardTCPClient_ or _MockEchoTCPServer_ to test _EchoTCPClient_
- Adds MockDaytimeTCPServer that uses the given clock to mock returned daytime string
- Adds DaytimeTCPClient integration tests

FYI
@garydgregory @kinow @sebbASF

garydgregory commented:
@jkbkupczyk 
Thank you for PR.
Please add a brief class level Javadoc to each new class.
Files should end in an empty line.

jkbkupczyk commented:
> @jkbkupczyk Thank you for PR. Please add a brief class level Javadoc to each new class. Files should end in an empty line.

Thank you @garydgregory for review :)
I added missing _Jacadocs_ and rebased onto master.

jkbkupczyk commented:
Hi @garydgregory, any new comments regarding this PR?

garydgregory commented on src/test/java/org/apache/commons/net/tftp/TFTPServer.java:
```diff
@@ -610,7 +610,7 @@ protected void finalize() throws Throwable {
 
     /**
-     * Gets the current value for maxTimeoutRetries
+     * Get the current value for maxTimeoutRetries
```
Don't change the style of the Javadoc please. A getter "Gets...".

jkbkupczyk replied:
I pushed that code by accident, and I already reverted it back. In pull #173 [(that part)](https://github.com/apache/commons-net/pull/173/files#diff-8cfe322813dd0689aac3ae713d40911dfba80a16e7bcad4f6a95d1106e4da5baR613) is fixed

garydgregory commented:
-1 as the PR stands today: The PR does not test anything new that isn't already tested.
Before and after this PR, the code coverage is the same at: 40% instructions / 31% branches.
Run `mvn clean site -Pjacoco` then open `target/site/jacoco/index.html`
Or am I missing something?

jkbkupczyk commented:
> -1 as the PR stands today: The PR does not test anything new that isn't already tested. Before and after this PR, the code coverage is the same at: 40% instructions / 31% branches. Run `mvn clean site -Pjacoco` then open `target/site/jacoco/index.html` Or am I missing something?

Hi, @garydgregory
I ran `mvn clean site -Pjacoco` both on `master` and `mock_tcp_test_server_daytimetcpclient_tests` and these are the results I've got:

**master**
![master](https://github.com/apache/commons-net/assets/69753899/f1a1ac44-1bff-4f92-8ef5-f56827fd5460)

**mock_tcp_test_server_daytimetcpclient_tests**
![mock_tcp_test_server_daytimetcpclient_tests](https://github.com/apache/commons-net/assets/69753899/3b65c9df-e7d7-4ebb-996a-1cdcd2cd2de6)

did you check `site/jacoco/org.apache.commons.net.daytime/DaytimeTCPClient.html`?

garydgregory commented:
@jkbkupczyk 
Well, it works for me now locally. Merged! Sorry about the delay.
